### ISSUE TITLE
Global swiftAddressBook variable is null when contact permission is denied

### ISSUE THREAD

**Issue by almilli** [SocialbitGmbH/SwiftAddressBook#72]
If you deny the contact permission for the app and then kill/restart the app, when it tries to construct the SwiftAddressBook instance it will init to null.  It looks like it's due to an error in ABAddressBookCreateWithOptions and how you are using it.

https://developer.apple.com/library/ios/documentation/AddressBook/Reference/ABAddressBookRef_iPhoneOS/#//apple_ref/c/func/ABAddressBookCreateWithOptions

The documentation says this:
```
On iOS 6.0 and later, if the caller does not have access to the Address Book database:

For apps linked against iOS 6.0 and later, this function returns NULL.

For apps linked against previous version of iOS, this function returns an empty read-only database.
```

Basically the constructor shouldn't rely on the address book creation and just fail miserably if it doesn't get it just yet.

**Comment by TAKeanice** [comment]
How wouöd you consider a proper failure? Just like a failable initializer, the failure here is signaled by the SwiftAddressBook being null. Since it is an optional variable, Swift forces you to anticipate that possibility.

**Comment by almilli** [comment]
The problem is it's a singleton that's only ever initialized once.  If the user goes in and changes their access it's never checked again until the app is killed and restarted.  You can't even call the request access method since it's on the singleton.  If you do using the ? syntax then you never get a completion callback, instead it just silently ignores it.  IMO, the variable should really never be null and it should react to the current state of the device whenever methods are called.  So if it hasn't created the reference to the internal address book it attempts to do so when needed.

**Comment by TAKeanice** [comment]
If one denies permission, the variable should definitely be null, and as far as I have tested, the app gets killed anyway when changing permissions! The access is requested correctly, as you can see in the sample app, but you are right, that's strange because it is on the nil object...

**Comment by TAKeanice** [comment]
I have checked again in the documentation. Seems like, before iOS6 (which is not relevant for us) the addressBook variable is not null, but empty before requesting access. After iOS6, the access is requested when the `ABAddressBookCreateWithOptions` function is called first. Thus, after creation, if the created addressBook is nil, the access was denied, otherwise the access was granted.
Since the app is restarted anyway, when the permissions have changed, I don´t see any problem here.

Maybe I should do some refactoring by removing the obsolete reqestAccess function.
